restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/util/RCUtils.java

```diff
@@ -173,4 +177,90 @@ public static void validateConnectionParms(HashMap<String, Object> parameters) t
       }
 
    }
+
+   public static boolean shouldUpdatePush(HashMap<String, Object> parameters) throws RCException{
+      if (!parameters.containsKey(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_FCM_SERVER_KEY) ||
+              parameters.get(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_FCM_SERVER_KEY) == null ||
+              TextUtils.isEmpty((String) parameters.get(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_FCM_SERVER_KEY))) {
+         throw new RCException(RCClient.ErrorCodes.ERROR_DEVICE_PUSH_NOTIFICATION_FCM_SERVER_KEY_MISSING);
+      }
+      return true;
+   }
+
+   public static boolean shouldRegisterForPush(HashMap<String, Object> parameters, StorageManagerInterface storageManagerInterface) throws RCException{
+      //check is there fcm server key, if not write a warning
+      if (parameters == null) {
+         return false;
+      }
+
+      if (!parameters.containsKey(RCDevice.ParameterKeys.PUSH_NOTIFICATIONS_FCM_SERVER_KEY) ||
```

Comment:
On this first part of shouldRegisterForPush() where you do validation checks on push-related parameters: doesn't it fit better in RCUtils.validateSettingsParms()? That way the user gets notified in the App synchronously and early that they have messed up the configuration before trying to do any signaling stuff. Wdyt?

Reply:
I think we should have it separate from signaling settings because of 2 things:
1. We can enable/disable push. When push is disabled, we will not check any of the push settings. So its easy skippable from the outside. If we put it in the validateSettingsParms we will need to add some logic there
2. In the previous comment of the PR ( Can we avoid and make this method return false if none of the push notification related parameters ....). I agree we should have it separate. Once again it will be more logic to add in validateSettingsParms instead of "outside".
I think we should have as little as possible business logic inside RCUtils wdyt?
I will not make changes until we agree what we will do on this. Lets talk tomorrow.

Reply:
1. But isn't there a flag in the parameters on whether push is enabled or not that we can easily use? Notice that we already have logic in Utils, so as long as the logic is simple we should be ok. It's good UX to do all the first level validations when the user expects them. And with this change we are splitting them to 2 places: one as a exception thrown in RCDevice.inititalize() and another as a result of the onInitialized(), which is a bit confusing. We can discuss more in daily if you want
2. Again simple validation logic should be ok in utils and all we 'll need I think is validate all push related parameters if push is enabled right?

Reply:
What we agree on:
1. RCUtils.validateSettingsParms() will extend with boolean param is push is changed or not
2. We will introduce new push param check in initialize() method. If its not validated we will throw an RCException